### DISCUSSION TITLE
socket.io client does not receive events

### DISCUSSION THREAD

**Discussion by bennetrr** [miguelgrinberg/python-socketio#1211] [Q&A]
Hi,
I have a Svelte application that is using the socket.io client, and a python application that is using your socket.io server with eventlet. The important code is in these files:

- https://github.com/bennetrr/weckpi/blob/dev/src/core/src/weckpi/core/main.py
- https://github.com/bennetrr/weckpi/blob/dev/src/web/src/lib/BackendConnection/ParameterCommunication.ts
- https://github.com/bennetrr/weckpi/blob/dev/src/web/src/lib/BackendConnection/WeckPiCoreConnection.ts

In my Svelte app I have stores that can be changed by the UI. These stores have listeners that send a `property-changed` event to the python app, where the event is processed. This part works, I have a music player that uses the MPV library and can control e.g. the volume with my Svelte app.

The same thing should also work in the other direction. I created an event listener on the MPV player that activates when the position in the current song changes. This listener should then emit the `property-change` event, so that the client updates the progress bar.

I can see the following messages in the python output:

```python
[16:31:47 | weckpi.core.main.socket | INFO] emitting event "property-change" to all [/]
[16:31:47 | weckpi.core.main.socket | INFO] WQkyYXTeEukf-enoAAAC: Sending packet MESSAGE data 2["property-change",{"prop":"music.position","value":0.15308763560090696}]
[16:31:47 | weckpi.core.main.socket | INFO] NiVDeple0FFaql-4AAAE: Sending packet MESSAGE data 2["property-change",{"prop":"music.position","value":0.15308763560090696}]
```

But I see nothing on the client side. Also `sio.onAny()` does nothing.

**Comment by miguelgrinberg** [comment]
wouldn't this be a problem in your client? The logs show that two connected clients were sent the `property-change` event. I think the first step in debugging the issue would be to determine why the client is dropping these events.

**Reply by bennetrr** [reply]
Should I open a discussion over there? They are not as quick with answering as you, tho :)

**Reply by miguelgrinberg** [reply]
I really don't know. My guess is that this is a bug in your application, not an issue with the socket.io client or server. Maybe ask in StackOverflow?

**Reply by bennetrr** [reply]
For reference: I opened a question at stackoverflow: https://stackoverflow.com/questions/76642710/socket-io-client-does-not-receive-events

I'll leave this discussion open until I get a solution. Thanks for your help so far!
